raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -281,25 +288,29 @@ library NettingChannelLibrary {
         bytes32 h;
         bytes32 hashlock;
 
-        (expiration, amount, hashlock) = decodeLock(locked_encoded);
+        // Check if caller_address is a participant and select her partner (for
+        // third party unlock see #541)
+        uint8 index = 1 - index_or_throw(self, caller_address);
```

Comment:
We already have this code. It was not added, just moved around. Check [here](https://github.com/raiden-network/raiden/blob/c245a10cbec985303280d7077b764bfb2881af62/raiden/smart_contracts/NettingChannelLibrary.sol#L299)